hudi-utilities/src/test/java/org/apache/hudi/utilities/sources/TestDFSSource.java

```diff
@@ -113,5 +114,66 @@ public void testJsonDFSSource() throws IOException {
     InputBatch<JavaRDD<GenericRecord>> fetch4 =
         jsonSource.fetchNewDataInAvroFormat(Option.of(fetch2.getCheckpointForNextBatch()), Long.MAX_VALUE);
     assertEquals(Option.empty(), fetch4.getBatch());
+
+    // 5. Extract from the beginning
+    InputBatch<JavaRDD<GenericRecord>> fetch5 = jsonSource.fetchNewDataInAvroFormat(Option.empty(), Long.MAX_VALUE);
+    assertEquals(10100, fetch5.getBatch().get().count());
+  }
+
+  @Test
+  public void testParquetDFSSource() throws IOException {
+    dfs.mkdirs(new Path(dfsBasePath + "/parquetFiles"));
+    HoodieTestDataGenerator dataGenerator = new HoodieTestDataGenerator();
+
+    TypedProperties props = new TypedProperties();
+    props.setProperty("hoodie.deltastreamer.source.dfs.root", dfsBasePath + "/parquetFiles");
+    ParquetSource parquetDFSSource = new ParquetDFSSource(props, jsc, sparkSession, schemaProvider);
+    SourceFormatAdapter parquetSource = new SourceFormatAdapter(parquetDFSSource);
+
+    // 1. Extract without any checkpoint => get all the data, respecting sourceLimit
+    assertEquals(Option.empty(), parquetSource.fetchNewDataInAvroFormat(Option.empty(), Long.MAX_VALUE).getBatch());
+    List<GenericRecord> batch1 = Helpers.toGenericRecord(dataGenerator.generateInserts("000", 100), dataGenerator);
+    Path file1 = new Path(dfsBasePath + "/parquetFiles", "1.parquet");
+    Helpers.saveParquetToDFS(batch1, file1);
+    assertEquals(Option.empty(), parquetSource.fetchNewDataInAvroFormat(Option.empty(), 10).getBatch());
```

Comment:
maybe need an assert that the size of 1.parquet must be larger than 10?

Reply:
Not sure if I understand correctly. We assert `1.parquet` is equal to 100 at line141. I think it achieves the same goal of `1.parquet` much be larger than 10. What do you think?

Reply:
Sorry for not express clearly. I think 1.parquet file length should be larger than 10(sourceLimit), otherwise `parquetSource.fetchNewDataInAvroFormat` would read data and thus batch is not empty, and lead to the faiure of `assertEquals(line 138)`. 
Of cause after 100 records written to 1.parquet, We can basically conclude that the file length is greater than 10. But I think it is better to assert the file length must larger than 10. WDYT?

Reply:
Gotcha. This is a very good point. Since `sourceLimit` was in `bytes` and we can not assert the number of the bytes by the number of records.